### DISCUSSION TITLE
What is the limit for the number of users that can be added to a session?

### DISCUSSION THREAD

**Discussion by RichieMcCaff** [microsoft/live-share-sdk#797] [Q&A] [answered]
I've been told the number is somewhere around 100 but I can't find any documentation on this.
Also for any user that joins after the limit is there a specific error that gets thrown for them?

**Comment by ryanbliss** [comment, marked as answer]
The limit is 100 due to a limit set by Azure Fluid Relay. It is documented in our [FAQ](https://learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/teams-live-share-faq). I will try and get confirmation on the specific error message. 

EDIT: The error thrown by Fluid can be found [here](https://github.com/microsoft/FluidFramework/blob/64b64a017146e63d464f47b1e2cf1418a773168d/server/routerlicious/packages/lambdas/src/nexus/connect.ts#L499). Message is `Too Many Clients Connected to Document`, status code is 429. Will update answer again once I confirm how this gets exposed via the SDK.

**Reply by ryanbliss** [reply]
@RichieMcCaff updated to include the error message and will update again once I get confirmation on how this gets exposed via the SDK. Our story for this isn't very robust, since such a small % of meetings are above 100 participants. I'll make sure to update the FAQ in the Live Share docs to include instructions for how to catch the error so users can be given instructions, and maybe add some more disclaimers in our docs so it is easier to find.

**Reply by jhodges10** [reply]
Is there any chance this limit could be raised higher with limited functionality, like a "one leader + many followers only" mode? I was really banking on this supporting up to ~500-1000 users (our department size) for internal usage!

**Reply by ryanbliss** [reply]
@jhodges10 unfortunately the limit exists within the Azure Fluid Relay backend and isn't configurable. We've been working with them to change this, but the cost is high since they'd have to put in general-purpose safeguards to protect their service from misuse. While it is possible, Azure approaches problems like this with quite a bit of sophistication and caution.

With that said, Microsoft has an internal version of Fluid backend deployed to the ODSP cloud. It is used by SharePoint-based apps like Loop, Whiteboard, etc. It has some differences in implementation details (e.g., auth, storage, etc.), but generally works very similarly. We have worked with them closely to understand what they look for in large session sizes, and for one partner we were able to get approval for >100 users. To get that approval, we had to do something similar to what you are suggesting.

All of this is to say that the problem is well understood. What Azure is looking at is how to put in service level constraints that enforce it is being used properly. That means A) knowing what redlines to put in place to ensure network health, and B) enforcing those redlines. They also want to ensure they have adequate E2E testing in place to ensure those enforcements are working as intended.

At this point it's a matter of "when" it will be done, not "if." The ask has plenty of visibility and weight, and each partner asking for it helps a lot with prioritization.

**Reply by jhodges10** [reply]
Thanks for the detailed response @ryanbliss! Consider Frame.io/Adobe another partner interested in having 100+ users supported, even in a limited capabilities mode.
